File: aro-hotel.UI/ClientApp/src/app/infrastructure/models/hotel.ts
import { IAddress } from "./IAddress";
import { IMultimedia } from "./Multimedia";
import { IRoom } from "./Room";

export interface IHotel {
  id?: number;
  name?: string;
  description?: string;
  address?: IAddress;
  phone?: string;
  multimedias?: IMultimedia[],
  rooms?: IRoom[],
  facilities?: string[],
  discount?: number;
}

File: aro-hotel.Domain/Model/Hotel.cs
namespace aro_hotel.Domain.Model;
public class Hotel : Entity
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Phone { get; set; }
    public int Discount { get; set; }

    public ICollection<Room> Rooms { get; set; }
    public Address Address { get; set; }

    public virtual ICollection<HotelFacilityXREF> HotelFacilityXREFs { get; set; }
    public virtual ICollection<HotelMultimediaXREF> HotelMultimediaXREFs { get; set; }
}
